src/services/api.js

```diff
@@ -42,19 +42,25 @@ export const getLocation = ({ id }) =>
       .then(result => result.data);
   });
 
-const updateResource = (path, { id, params }) =>
+const updateResource = ({pathPrefix, method, pathSuffix}, { id, params }) =>
   Amplify.Auth.currentAuthenticatedUser().then((user) => {
     const idJwtToken = user.signInUserSession.getIdToken().getJwtToken();
 
+    //construct the path
+    const pathComponents = [config.baseApi, pathPrefix, id];
+    if(pathSuffix) pathComponents.push(pathSuffix)
+    const url = pathComponents.join('/');
+
     return axios.request({
-      url: `${config.baseApi}/${path}/${id}`,
-      method: 'patch',
+      url,
+      method,
       data: params,
       headers: {
         Authorization: idJwtToken,
       },
     });
   });
 
-export const updateLocation = updateResource.bind(this, 'locations');
-export const updatePhone = updateResource.bind(this, 'phones');
+export const updateLocation = updateResource.bind(this, { pathPrefix: 'locations', method: 'patch' });
+export const updatePhone = updateResource.bind(this, { pathPrefix: 'phones', method: 'patch' });
+export const createPhone = updateResource.bind(this, { pathPrefix: 'locations', method: 'post', pathSuffix: 'phones' });
```

Comment:
Personally I would probably favor simplicity over DRYness in this case (as we're getting close to the point `updateResource` will need to take as many args as `axios.request` :P), but doesn't make much difference either way.